### DISCUSSION TITLE
splitMeshRegions before import the mesh, how it's possible in Windows ?

### DISCUSSION THREAD

**Discussion by andprad74** [nextfoam/baram#64] [General]
Dear all,
in BaramCFD i use mainly the solver chtMultiRegionfoam  with the mesh made in the format fluent .msh or i-DEAS .unv in Windows.
The software is able to convert the mesh but not  split the mesh in different regions.
How i can by pass that problem by using the Windows powershell for instance?
Actually i made that operation in a ubuntu 18 virtual machine and baram 6.4.
Thanks in adwance and also thanks for that amazing tool.
Andrea.

**Comment by jiban** [comment]
@andprad74
To understand your needs precisely, could you upload one of your mesh files here?

**Comment by andprad74** [comment]
Dear jiban,
first of all thanks for your suppport.
The test mesh is made with salome and it's a pipe with water inside. The 2 groups of volume are made as  all the faces groups for the boundary condition setup
The unv file imported in Baram is converted, please see the picture, now it's necessary to split the mesh in the 2 different regions in according to the 2 cell set, tube and water.
In baram 6.4 that operation was made in 2 step. First I import and convert the mesh in the openfoam version inside BARAM and after, inside the BARAM Multiregionsolver i made the splitting. The operation conversion and splitting made inside the BARAM Multiregionsolver doesn't work, i don't know if it's a bug ...
 
Inside windows i have try to setup the path for the openfoam .dll. libraries  (../baram/solver/openfoam/lib)  and for the splitMeshRegions  command (please see the picture) and perform the splitting  operation at command line as superuser but doesn't work.

In my point of view in the tool bar Mesh, in additional to the command "rotate", "scale" and "traslate", will be great to have available the splitting operation or something else in the cell zone conditions. 

Thanks in advance. 
andrea.


![dos_windows_test](https://github.com/nextfoam/baram/assets/131922029/610edae2-66ff-4b97-a8e5-ef0a536b28c9)
![salome](https://github.com/nextfoam/baram/assets/131922029/8a81bc6e-2a62-4826-a63b-580f20762a4c)
![buondary](https://github.com/nextfoam/baram/assets/131922029/12a49f1d-63fd-444d-a253-6a390853633b)
![cell_zone](https://github.com/nextfoam/baram/assets/131922029/775becee-0d37-4d2b-ab72-ead01915fd61)




[Mesh_for_evaluation.zip](https://github.com/nextfoam/baram/files/14815082/Mesh_for_evaluation.zip)

**Comment by kdg5424** [comment]
Hi @andprad74 . This is Donggyu in NEXTFOAM. Thank you for your attention in BARAM.
Why don't you use this script on Salome? It's the script to export the mesh for baramFlow.
While you export the mesh using the script, we will check and test your mesh.
Thank you.

[salomeToBaram.zip](https://github.com/nextfoam/baram/files/14818213/salomeToBaram.zip)

**Comment by andprad74** [comment]
Dear Donggyu thanks  for your fast reply.
I have import the script from the menu FILE\LOAD SCRIPT..., I don't know if it's correct. If i will select the 3d part the export will be a STl file in  salome directory, nothing more. If i preselect the mesh the result is an error. The salome version is the 9.11.0 for windows.
In any case in office i use the ANSYS tool for made the mesh, the final result is a fluent mesh in ASCII that i need to import in baram and split.
The conversion is ok but the splitting problme remain.
Thanks.
Andrea

**Comment by jiban** [comment]
@andprad74 
You seem familiar with `splitMeshRegions` command.
OpenFOAM commands embedded in BARAM can be used in Windows *Command Prompt* if `path` is properly configured.
Following is my configuration.
```cmd
set WPD=C:\Users\jakej\Documents\workspace\baram\solvers\openfoam
set MGW=C:\Users\jakej\Documents\workspace\baram\solvers\mingw64
set WM_PROJECT_DIR=%WPD%
path %MGW%\bin;%MGW%\lib;%WPD%\bin;%WPD%\lib;%WPD%\lib\msmpi;%Path%
```

And here is a sample execution on Window *Command Prompt*.

![image](https://github.com/nextfoam/baram/assets/1674177/b4558d54-e345-40a4-87af-d03cbdc29ded)

**Comment by andprad74** [comment]
Thanks a lot Jake now i can convert and split the mesh before to open a project in Baram.

Dear Donggyu can you explain better how works the script salomeTobaram? i'm intersted because the .unv file doesn't export the pyramid elements and sometime i made some parametric mesh by using Yacs in salome. 
In case of congruent mesh salome use the pyramid element between the tetra and hexa element.
I would like to create parametric models from Salome to Openfoam managed by YACS.
Thanks a lot.
Andrea

**Comment by kdg5424** [comment]
Hi @andprad74 . After I ask script developer how to use it exactly, let me tell you.
I used the script as you did.
For just in case, I will make it sure.

**Comment by kdg5424** [comment]
Hi @andprad74 . I asked script developer.
The way how you did is correct.
After you selected the mesh, click File/Load script. Then it would export the mesh.
Thanks.

**Comment by andprad74** [comment]
Dear Donggyu,
thanks a lot.
A.

**Comment by kdg5424** [comment]
Glad to hear that.
If you're ok, let me close the issue.

**Comment by andprad74** [comment]
yes you can close the the issue
thanks.
A.

**Reply by kdg5424** [reply]
Thank you. If you have any problem with BARAM, please contact us.
